File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque


def input():
    return sys.stdin.readline().rstrip()


n = int(input())
queue = deque()
commands = {
    'pop': lambda: print(-1) if not queue else print(queue.popleft()),
    'size': lambda: print(len(queue)),
    'empty': lambda: print(1) if not queue else print(0),
    'front': lambda: print(-1) if not queue else print(queue[0]),
    'back': lambda: print(-1) if not queue else print(queue[-1]),
}

for _ in range(n):
    command = input().split()
    if command[0] == 'push':
        queue.append(int(command[1]))
    else:
        commands[command[0]]()

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <string>
#include <queue>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int n, x;
	cin >> n;
	string cmd;
	queue<int> q;
	while (n--) {
		cin >> cmd;
		if (cmd == "push") {
			cin >> x;
			q.push(x);
			continue;
		}
		else if (cmd == "pop") {
			if (q.empty()) cout << -1;
			else {
				cout << q.front();
				q.pop();
			}
		}
		else if (cmd == "size") {
			cout << q.size();
		}
		else if (cmd == "empty") {
			cout << (q.empty() ? 1 : 0);
		}
		else if (cmd == "front") {
			if (q.empty()) cout << -1;
			else cout << q.front();
		}
		else if (cmd == "back") {
			if (q.empty()) cout << -1;
			else cout << q.back();
		}
		cout << '\n';
	}
}